test/Mono.Linker.Tests.Cases/Reflection/ObjectGetTypeLibraryMode.cs

```diff
@@ -0,0 +1,154 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Text;
+using Mono.Linker.Tests.Cases.Expectations.Assertions;
+using Mono.Linker.Tests.Cases.Expectations.Helpers;
+using Mono.Linker.Tests.Cases.Expectations.Metadata;
+
+namespace Mono.Linker.Tests.Cases.Reflection
+{
+	[SetupLinkerArgument ("-a", "test.exe", "library")]
+	[ExpectedNoWarnings]
+	[KeptMember (".ctor()")]
+	public class ObjectGetTypeLibraryMode
+	{
+		public static void Main ()
+		{
+			BasicAnnotationWithNoDerivedClasses.Test ();
+			BasicNoAnnotationWithNoDerivedClasses.Test ();
+		}
+
+		[Kept]
+		class BasicAnnotationWithNoDerivedClasses
+		{
+			[Kept]
+			[KeptAttributeAttribute (typeof (DynamicallyAccessedMembersAttribute))]
+			[DynamicallyAccessedMembers (DynamicallyAccessedMemberTypes.PublicMethods)]
+			public interface IBasicAnnotatedInterface
+			{
+			}
+
+			[Kept]
+			[KeptMember (".ctor()")]
+			[KeptInterface (typeof (IBasicAnnotatedInterface))]
+			class ClassImplementingAnnotatedInterface : IBasicAnnotatedInterface
+			{
+				[Kept]
+				public void UsedMethod () { }
+				[Kept] // The type is not sealed, so trimmer will apply the annotation from the interface
+				public void UnusedMethod () { }
+			}
+
+			[Kept]
+			static void TestInterface (ClassImplementingAnnotatedInterface classImplementingInterface)
+			{
+			}
```

Comment:
Nit: It would be a little bit cleaner to just new-up the instance inside this method - instead of passing it as a parameter (as is, this method doesn't nothin, even during marking it won't do anything interesting).